packages/datadog_flutter_plugin/lib/src/rum/rum_user_action_detector.dart

```diff
@@ -283,6 +286,14 @@ class _RumUserActionDetectorState extends State<RumUserActionDetector> {
     String? elementName;
     bool searchForBetter = false;
     bool searchForText = true;
+
+    final customWidgetDetected =
+        this.widget.customGestureDetector?.call(widget);
+    if (customWidgetDetected != null) {
+      elementName = customWidgetDetected.elementName;
+      searchForBetter = customWidgetDetected.searchForBetter;
+      searchForText = customWidgetDetected.searchForText;
+    }
     if (widget is ButtonStyleButton) {
```

Comment:
This should probably be `else if`, otherwise the custom widget information will be overwritten if the widget happens to also be a `ButonStyleButton`.

Reply:
Agreed, but that doesn't mean a custom component couldn't inherit from a builtin component, in which case it would match both the custom logic and the checks below would overwrite it